scipy/integrate/_ivp/tests/test_ivp.py

```diff
@@ -510,6 +543,39 @@ def event_rational_3(t, y):
         assert np.allclose(sol_rational(res.t_events[1][0]), res.y_events[1][0], rtol=1e-3, atol=1e-6)
         assert np.allclose(sol_rational(res.t_events[2][0]), res.y_events[2][0], rtol=1e-3, atol=1e-6)
 
+        res = solve_ivp(fun_rational, [8, 5], [4/9, 20/81], method=method,
+                        events=(event_rational_1, event_rational_2,
+                                event_rational_4),
+                        dense_output=True, max_step=0.5)
+        assert_equal(res.status, 1)
+        assert_equal(res.t_events[0].size, 0)
+        assert_equal(res.t_events[1].size, 1)
+        assert_equal(res.t_events[2].size, 2)
+        assert_(7.3 < res.t_events[1][0] < 7.7)
+        assert_allclose(res.t_events[2][0], 7)
+        assert_allclose(res.t_events[2][1], 6)
+
+        assert_equal(res.y_events[0].shape, (0,))
+        assert_equal(res.y_events[1].shape, (1, 2))
+        assert_equal(res.y_events[2].shape, (2, 2))
+        assert np.isclose(
+            event_rational_2(res.t_events[1][0], res.y_events[1][0]), 0)
+        assert np.isclose(
+            event_rational_4(res.t_events[2][0], res.y_events[2][0]), 0)
+        assert np.isclose(
+            event_rational_4(res.t_events[2][1], res.y_events[2][1]), 0)
+
+        # Also test that termination by event doesn't break interpolants.
+        tc = np.linspace(res.t[-1], res.t[0])
+        yc_true = sol_rational(tc)
+        yc = res.sol(tc)
+        e = compute_error(yc, yc_true, 1e-3, 1e-6)
+        assert_(np.all(e < 5))
+
+        assert np.allclose(sol_rational(res.t_events[1][0]), res.y_events[1][0], rtol=1e-3, atol=1e-6)
+        assert np.allclose(sol_rational(res.t_events[2][0]), res.y_events[2][0], rtol=1e-3, atol=1e-6)
+        assert np.allclose(sol_rational(res.t_events[2][1]), res.y_events[2][1], rtol=1e-3, atol=1e-6)
```

Comment:
Won't `assert_allclose()` tend to give better error handling/mismatch feedback when the test fails instead of asserting on the `bool` return value of `allclose()`? May be similar with use of i.e., `assert_array_less()`  in a case or two above.

there are a few other cases above where older things like `assert_` could be replaced with modern alternatives for *new* tests I suppose

Reply:
I have already done some updates on another [PR](https://github.com/scipy/scipy/pull/14552/files), but it is likely that the present PR will be merged first considering reviewing time. Should I apply the same updates here? Additionally, does `assert_allclose()` make sense in the context of two floats (as opposed to two ndarrays of floats)? Or is there another preferred approach?

Reply:
You can still use `assert_allclose` on scalars, and that is preferred over `assert np.allclose`. no need to change existing code in the tests to a new format unless you're changing something else about the tests.